packages-ts/gauntlet-terra-contracts/src/commands/contracts/migration/index.ts

```diff
@@ -0,0 +1,84 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { TerraCommand, TransactionResponse } from '@chainlink/gauntlet-terra'
+import { AccAddress, MsgMigrateContract } from '@terra-money/terra.js'
+import { CONTRACT_LIST } from '../../../lib/contracts'
+
+type CommandInput<MigrationMessage> = {
+  contract: string
+  newCodeId: number
+  migrationMsg: MigrationMessage
+}
+
+export type MigrationMessageMaker<MigrationMessage> = (flags, args) => MigrationMessage
+
+export const makeMigrationCommand = <MigrationMessage>(
+  contract: CONTRACT_LIST,
+  makeMigrationMessage: MigrationMessageMaker<MigrationMessage>,
+) => {
+  return class MigrateContract extends TerraCommand {
```

Comment:
There is a lot of smart programming going on here that makes things harder to follow. I'd suggest defining a class as a top-level type, and then you could have a static constructor on it or a separate factory class/fn.

Not really fond of defining a (large) class in a closure like this.

Reply:
Yep I see your point, this makes the Class inaccessible from outside. Let me fix it